packages/abi-typegen/src/templates/contract/factory.hbs

```diff
@@ -15,15 +15,20 @@ export class {{capitalizedName}}Factory extends ContractFactory {
     super(bytecode, {{capitalizedName}}.abi, accountOrProvider);
   }
 
+  deploy<TContract extends Contract = Contract>(
+    deployOptions?: DeployContractOptions
+  ): Promise<DeployContractResult<TContract>> {
+    return super.deploy({
+      storageSlots: {{capitalizedName}}.storageSlots,
```

Comment:
We add storage slots on the contract while they're relevant only for deployment purposes, so it seems to me that we should move them to the factory. That would technically constitute a breaking change because we'd remove the static `MyTypegenContract.storageSlots` field, so we should add this as a TODO to
- #3119

We could add a deprecation note on the `MyTypegenContract.storageSlots` field.

This move of storage slots to from contract to contract factory can also be done as part of #3032.

Reply:
I agree that moving this variable makes sense. I've added the following [comment](https://github.com/FuelLabs/fuels-ts/issues/3032#issuecomment-2337740796), on the issue.

I'd personally shy away from the deprecation notice (for now).